src/Http/Ownership/SuggestOwnershipsRequestHandler.php

```diff
@@ -0,0 +1,148 @@
+    ) {
+        $this->resultsGenerator = new ResultsGenerator(
+            $searchService,
+            new Sorting('modified', 'desc')
+        );
+        $this->offerRepository = $offerRepository;
+        $this->currentUser = $currentUser;
+        $this->userIdentityResolver = $userIdentityResolver;
+        $this->organizerRepository = $organizerRepository;
+        $this->ownershipSearchRepository = $ownershipSearchRepository;
+        $this->organizerIDParser = $organizerIDParser;
+        $this->userPermissionsRepository = $userPermissionsRepository;
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $this->guardAccess();
+
+        $queryParams = new QueryParameters($request);
+
+        $queryParams->guardRequiredEnum('itemType', [ItemType::organizer()->toString()]);
+
+        $id = $this->currentUser->getId();
+        $user = $this->userIdentityResolver->getUserById($id);
+
+        $query = (new SuggestOwnershipsSapiQuery($user))->toString();
+
+        $activeOwnerships = $this->ownershipSearchRepository
+            ->search(new SearchQuery([
+                new SearchParameter('ownerId', $user->getUserId()),
+            ]))
+            ->filter(self::activeOwnerships());
+
+        /**
+         * A map to deduplicate returned organizers
+         * @var array<string, array> $idToOrganizerMap
+         */
+        $idToOrganizerMap = [];
+
+        /**
+         * @var ItemIdentifier $result
+         */
+        foreach ($this->resultsGenerator->search($query) as $result) {
+            $offerType = OfferType::fromCaseInsensitiveValue($result->getItemType()->toString());
+            $id = $result->getId();
+
+            $offer = $this->offerRepository->fetch($offerType, $id)->getAssocBody();
+
+            $organizerId = $this->organizerIDParser->fromUrl(new Url($offer['organizer']['@id']))->toString();
+
+            $ownershipsForOrganizer = $activeOwnerships->filter(fn (OwnershipItem $item) => $item->getItemId() === $organizerId);
+
+            // don't suggest item that already has active ownership
+            if (!$ownershipsForOrganizer->isEmpty()) {
+                continue;
+            }
+
+            // don't suggest duplicate items
+            if (isset($idToOrganizerMap[$organizerId])) {
+                continue;
+            }
+
+            $organizer = $this->organizerRepository->fetch($organizerId)->getAssocBody();
+
+            $idToOrganizerMap[$organizerId] = $organizer;
+        }
+
+        return new JsonLdResponse(array_values($idToOrganizerMap));
+    }
+
+    private static function activeOwnerships(): callable
+    {
+        return function (OwnershipItem $item): bool {
+            $state = new OwnershipState($item->getState());
+            return OwnershipState::requested()->sameAs($state) || OwnershipState::approved()->sameAs($state);
+        };
+    }
+
+    private function guardAccess(): void
```

Comment:
I think this should use `organizer_permission_voter` like `OwnershipStatusGuard` does because the organizer ID is an important parameter to decide if the current user has access to the current organisation.

Reply:
I'm wrong, I don't even think there is a check needed.